### PR TITLE
Feat(event): add newrelic tracing to object

### PR DESCRIPTION
**Pull request by johndy2742**
### Changes Made

- add newrelic tracing to object

### Purpose

- provides real-time visibility into the health and performance of the app

### How has this been tested
- new relic monitoring

image

### Related Links

https://chat.digitalservice.id/boards/team/nii5pbyw1idwxrxjd8bp56mtkw/bte55h31wfjgp9xgry174ur3mae/vbyyag4aadi8ejgca6gtqzs55gc

Project: Event Management System
Participants: @johndy2742, @iqbal167, @alqinsidev, @MNAlfiansyah , @ristianaditya

### CODE REVIEW

File: service/src/app/app.go [hunk 1]
```diff
@@ -42,6 +44,15 @@ func Init() (*App, error) {
 	masterDB := InitPgsqlMaster(ctx, appConfig)
 	slaveDB := InitPgsqlSlave(ctx, appConfig)
 
+	nrApp, err := newrelic.NewApplication(
+		newrelic.ConfigAppName("eventManagement"),
+		newrelic.ConfigLicense("eu01xx7cbed379a22ba3d1760b466361FFFFNRAL"),
```
Comment (iqbal167) [inline service/src/app/app.go:49]: use environment variable

Comment (iqbal167) [reply service/src/app/app.go:49]: set this env in config
NEW_RELIC_APP_NAME=newrelic-labs
NEW_RELIC_LICENSE=
NEW_RELIC_CONFIG_APP_LOG_FORWARDING_ENABLED=true
NEW_RELIC_CONFIG_DISTRIBUTED_TRACER_ENABLED=true
NEW_RELIC_RECORD_PANICS=true

File: service/src/app/app.go [hunk 2]
```diff
@@ -42,6 +44,15 @@ func Init() (*App, error) {
 	masterDB := InitPgsqlMaster(ctx, appConfig)
 	slaveDB := InitPgsqlSlave(ctx, appConfig)
 
+	nrApp, err := newrelic.NewApplication(
```
Comment (iqbal167) [inline service/src/app/app.go:47]: move to a separate file (newrelic.go)  create init function and add struct for new relic config struct

File: service/src/app/app.go [hunk 3]
```diff
@@ -42,6 +44,14 @@ func Init() (*App, error) {
 	masterDB := InitPgsqlMaster(ctx, appConfig)
 	slaveDB := InitPgsqlSlave(ctx, appConfig)
 
+	if err != nil {
```
Comment (iqbal167) [inline service/src/app/app.go:47]: what's about this handle error?

File: service/src/app/newrelic.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"github.com/newrelic/go-agent/v3/newrelic"
+	"github.com/spf13/viper"
+)
+
+var nrApp *newrelic.Application
+
+func InitNewRelic() error {
+	var err error
+	nrApp, err = newrelic.NewApplication(
```
Comment (iqbal167) [inline service/src/app/newrelic.go:12]: please use config for get env variable in config.go file

File: service/src/modules/object/usecase/create_object.go [hunk 5]
```diff
@@ -4,13 +4,17 @@ import (
...
 func (uc *Usecase) CreateObject(ctx context.Context, obj request.Object) (*request.Object, error) {
+	txn := newrelic.FromContext(ctx)
+	usecaseSegment := txn.StartSegment("CreateObjectUsecase")
+
 	createdObj, err := uc.repo.CreateObject(ctx, obj)
 	if err != nil {
 		return nil, err
 	}
-
+	usecaseSegment.End()
```
Comment (iqbal167) [inline service/src/modules/object/usecase/create_object.go:18]: why is this end of a transaction set without deferring?

File: service/src/modules/object/endpoint/create_object.go [hunk 6]
```diff
@@ -11,6 +11,8 @@ import (
 )
 
 func (e *Endpoint) CreateObject(ctx context.Context, objData request.Object) (interface{}, error) {
+	endpointSegment := e.app.GetNewRelic().StartSegment(ctx, "CreateObjectEndpoint")
```
Comment (iqbal167) [inline service/src/modules/object/endpoint/create_object.go:14]: you can init the newRelicApp in the endpoint.go file first, and just call the method

File: service/src/modules/object/repository/object.go [hunk 7]
```diff
@@ -15,6 +15,17 @@ import (
 )
 
 func (r *Repository) CreateObject(ctx context.Context, obj request.Object) (request.Object, error) {
+	datastoreSegment := r.app.GetNewRelic().StartDatastoreSegment(
```
Comment (iqbal167) [inline service/src/modules/object/repository/object.go:18]: please init new relic app first in `repository.go`

File: service/src/modules/object/endpoint/create_object.go (hunk 6 above)
Comment (johndy2742) [reply service/src/modules/object/endpoint/create_object.go:14]: is there a particular reason to init the new relic first in the endpoint kang?

**Review by iqbal167 — APPROVED** [review]

**Comment by johndy2742** [comment]
please evaluate my approach kang @iqbal167 for this problem https://github.com/jabardigitalservice/event-management-system/pull/85/commits/2311917f7abe84109b7c5dd7c98d22e612dd6c3b

**Review by iqbal167 — APPROVED** [review]

**Pull request merged**